### DISCUSSION TITLE
v What is the support for wasm?

### DISCUSSION THREAD

**Discussion by Jengro777** [vlang/v#16406] [Ideas]
v What is the support for wasm?

**Comment by JalonSolov** [comment]
Compile to C, then use emscripten or similar to create the WASM code.

**Comment by sonygod** [comment]
~~1. before you start, you need to install [libgc ](https://zoomadmin.com/HowToInstall/UbuntuPackage/libgc-dev)~~

   

   ~~2. read [this ](https://github.com/vlang/v/tree/master/examples/call_v_from_c) tutorial  compile v to c. remember  add **-lgc**~~
 


  ~~3. and wasm tutorials  is [here](https://developer.mozilla.org/en-US/docs/WebAssembly/C_to_wasm)~~

from discord vlang  wasm 

spytheman — 09/23/2022

yeah, if you have https://emscripten.org/docs/getting_started/downloads.html installed, 

try: 
                      ./v -os wasm32_emscripten examples/2048/

it still needs a small fix, since it will generate examples/2048/2048 instead of examples/2048/2048.js, so you would have to rename that 

file manually afterwards, but other than that, it should work more or less ok

**Reply by Jengro777** [reply]
When can I support the command like rust: Cargo build -- target -- wasm32-wasi xxx.wasm?

For example: v - b -- target -- wasm32-wasi xxx.wasm

**Reply by JalonSolov** [reply]
`v -b wasm xxx.v`

Default output is WASI.

You can also be explicit with

`v -b wasm -os wasi xxx.v`

If you want WASM for the browser, use

`v -b wasm -os browser xxx.v`

You'll have to come up with your own html file to load it, or copy the one from `examples/wasm/mandelbrot/mandelbrot.html`

**Reply by Jengro777** [reply]
When v is compiled as. wasm, does it use GC or autofree memory management by default?
